### ISSUE TITLE
Question about batch size

### ISSUE THREAD

**Issue by michael-conrad** [DigitalPhonetics/IMS-Toucan#19]
I'm thinking about increasing the batch size to try and increase the training rate. My current ETA is 42 hours remaining.

I'm currently at ~50% GPU usage with 11,233 MiB on a GTX 3090 with 24,259 MiB.

Any considerations to take into account?

**Comment by Flux9665** [comment]
Because we count in steps and not in epochs, increasing the batchsize would make it go slower rather than faster. But since a larger batchsize leads to more stable gradient estimates, the actual learning progress made can be quicker. In general, the default amounts of steps I specified for most models is way too high. More steps generally don't hurt, but once you stop seeing improvements after like 10k steps in the progress plots, you can stop the run. It's a bit different for every dataset used, so I just used a rather high step count as default.

**Comment by michael-conrad** [comment]
10k, not 100K ?

That would definitely speed up my train/test cycle by a huge amount. Especially testing with removing lower quality data sets.

Exactly how to do I read the progress plots?

Right now I'm at step 150,000 with a total loss of 3.289 with an elapsed time of almost 23 hours.

I've been letting it run as the loss keeps going down in a consistent fashion.

**Comment by Flux9665** [comment]
By 10k steps I mean if over a time of 10k steps the progress plot does not significantly change, which for me usually happens at around 80k to 190k steps, depending on the dataset.

The progress plot is a spectrogram. The x axis is the time axis, the y axis is the frequency and the color is the amplitude. So you can see the amplitude of a frequency at a given time. What you're looking for are clear formants over time (harmonics near the resonance frequencies of the vocal tract, which are specific for every phoneme). So you want to see darker horizontal lines which are not totally flat in the lower part of the plot. The cleaner the data is, the more formants can generally be captures by the synthesis (more lines, also in the higher frequency regions), so if you have many such horizontal line, that indicates both good data and a well trained model. The lowest 2 formants are enough for a phoneme to be intelligible.

**Comment by michael-conrad** [comment]
Does this reduced iteration count also apply to the aligner training?

**Comment by Flux9665** [comment]
yes, however the aligner only needs a few thousand steps to get to near peak performance anyways. The massively multilingual aligner pretraining pipeline is only there so that the aligner can be used for finetuning on any new dataset, so I wanted it to have very good performance and to be very general. If you have a closed set of training datasets and you train the aligner on those anyways, then there's not much reson to go beyond 50k steps I think
